doc/release/upcoming_changes/template.rst

```diff
@@ -1,7 +1,4 @@
-{% set title = "NumPy {} Release Notes".format(versiondata.version) %}
-{{ "=" * title|length }}
-{{ title }}
-{{ "=" * title|length }}
+**Content from release note snippets in doc/release/upcoming_changes:**
```

Comment:
The setup makes sense to me, but besides it being done for 2.0, running things multiple time is just painful with or without this, since the sections will get messy.

So I think this line should be in the release notes file, the title I am happy to remove.  That way, when 2.0 is branched we create  new file with:

```
NumPy 2.1 release notes
=======================

.. inlcude: filename
```
and add this line if someone actually generates it for some reason intermittendly.

Reply:
Updated, and fixed the template for new release notes too.